### PR TITLE
Add support for postgres schema

### PR DESCRIPTION
**Pull request by nuradiyana**
Hi. this is PR for #235

In this PR i introduce new config called `using_schema_connection` with default value is `false`. I tired to get `DatabaseManager` base on `driver` so i using this config to check if should change schema instance of database.

What do you think?

### CODE REVIEW

**Comment by stancl** [comment]
The PR looks awesome, great job! I'll review it in depth later today/tomorrow and suggest some minor changes.

Thanks a lot!

File: tests/DatabaseSchemaManagerTest.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        $old_connection_name = app(\Illuminate\Database\DatabaseManager::class)->connection()->getName();
+        
+        tenancy()->init('test.localhost');
+
+        app(\Stancl\Tenancy\DatabaseManager::class)->reconnect();
+        
+        $new_connection_name = app(\Illuminate\Database\DatabaseManager::class)->connection()->getName();
+
+        // $this->assertSame($old_connection_name, $new_connection_name);
+        $this->assertNotEquals('tenant', $new_connection_name);
```
Comment (stancl) [inline tests/DatabaseSchemaManagerTest.php]: What's the reason for this?

**Review by stancl — COMMENTED** [review]
I'll open a PR to your repo that renames a few things. Otherwise this looks really good, great job :+1:

File: tests/DatabaseSchemaManagerTest.php (hunk 1 above)
Comment (nuradiyana) [reply tests/DatabaseSchemaManagerTest.php]: This test is copy from `DatabaseManagerTest` except this test `make_sure_using_schema_connection`, must i remove this?

Comment (stancl) [reply tests/DatabaseSchemaManagerTest.php]: I resolved this in my PR to your repo.

**Comment by stancl** [comment]
My PR: https://github.com/NoorAdiana/tenancy/pull/1

**Comment by stancl** [comment]
Tomorrow I'll send a PR that moves the `tenancy.separate_by` config to `tenancy.database.separate_by` and hopefully resolves the migration issue.

When that's fixed, I'll merge.

**Comment by stancl** [comment]
My PR to your repo fixes some things but the migration issue still persists.

**Comment by stancl** [comment]
If #289 is merged, the schema manager should probably implement `CanSetConnection` (w/ the new method signature).

**Comment by nuradiyana** [comment]
Sory for long long time response, i have a commit for fixing testing and i hope this enough to merge.

File: tests/DatabaseSchemaManagerTest.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        $database2 = config('database.connections.' . config('database.default') . '.database');
+
+        $this->assertSame($database1, $database2);
+        $this->assertNotSame($schema1, $schema2);
+    }
+
+    /** @test */
+    public function schemas_are_separated()
+    {
+        // copied from DataSeparationTest
```
Comment (morloderex) [inline tests/DatabaseSchemaManagerTest.php:95]: Based of your comment maybe we should create a trait for these common database tests. So we are sure that al behaves the same?

File: src/DatabaseManager.php [hunk 3]
```diff
@@ -101,16 +102,18 @@ public function createTenantConnection($databaseName, $connectionName)
...
 
     /**
      * Get the name of the connection that $connectionName should be based on.
      *
      * @param string $connectionName
-     * @return string
+     * @return string|null
      */
-    public function getBaseConnection(string $connectionName): string
+    public function getBaseConnection(string $connectionName): ?string
```
Comment (stancl) [inline src/DatabaseManager.php]: These changes should be reverted.

File: tests/DatabaseSchemaManagerTest.php (hunk 2 above)
Comment (stancl) [reply tests/DatabaseSchemaManagerTest.php:95]: It's pretty simple code and I value tests being explicit more than I value them being clean.

File: src/DatabaseManager.php (hunk 3 above)
Comment (nuradiyana) [reply src/DatabaseManager.php]: Done

File: src/TenantDatabaseManagers/PostgreSQLSchemaManager.php [hunk 4]
```diff
@@ -4,18 +4,25 @@
...
-use Illuminate\Database\DatabaseManager as IlluminateDatabaseManager;
+use Stancl\Tenancy\Contracts\Future\CanSetConnection;
+use Illuminate\Database\DatabaseManager;
 use Stancl\Tenancy\Contracts\TenantDatabaseManager;
 
-class PostgreSQLSchemaManager implements TenantDatabaseManager
+class PostgreSQLSchemaManager implements TenantDatabaseManager, CanSetConnection
 {
-    /** @var \Illuminate\Database\Connection */
+    /** @var Connection */
```
Comment (stancl) [inline src/TenantDatabaseManagers/PostgreSQLSchemaManager.php]: To make tenant creation queueable, we can't store the `Connection` instance here, but rather a string and resolve it to a connection using a method call - similarly to https://github.com/stancl/tenancy/blob/2.x/src/TenantDatabaseManagers/MySQLDatabaseManager.php. I'll add this change later today.

**Review by stancl — COMMENTED** [review]
I have no idea how this fixed the issues with migrations, but tests pass now so I'm happy.

**Comment by stancl** [comment]
Thanks everyone!

**Pull request merged**